binary_search_tree/tree.py

```diff
@@ -14,35 +17,96 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(1)
```

Comment:
Unfortunately as nodes are added to the tree, new nodes will have to be added deeper and deeper, meaning that this doesn't run in the constant time that O(1) implies. Assuming a balanced tree, the complexity will be O(log(n)). 

For example, if there are 7 nodes in a perfectly balanced tree, there would be 1 on top, 2 on the next, and 4 on the next. To add an 8th node will require recursing 3 times to add the new one on the new fourth level. And log_2(8) is 3, since 2^3 is 8.

Reply:
Your space complexity calculation is fine, however.